docs/source/tutorials/gateware_usb_device_03.rst

```diff
@@ -0,0 +1,301 @@
+                                    m.next = "HANDLE_SET_FPGA_LEDS"
+                                with m.Case(self.VENDOR_GET_USER_BUTTON):
+                                    m.next = "HANDLE_GET_USER_BUTTON"
+
+                    with m.State("HANDLE_SET_FPGA_LEDS"):
+                        # take ownership of the interface
+                        m.d.comb += interface.claim.eq(1)
+
+                        # if we have an active data byte, set the FPGA LEDs to the payload
+                        with m.If(interface.rx.valid & interface.rx.next):
+                            m.d.usb += fpga_leds.eq(interface.rx.payload[0:6])
+
+                        # once the receive is complete, respond with an ACK
+                        with m.If(interface.rx_ready_for_response):
+                            m.d.comb += interface.handshakes_out.ack.eq(1)
+
+                        # finally, once we reach the status stage, send a ZLP
+                        with m.If(interface.status_requested):
+                            m.d.comb += self.send_zlp()
+                            m.next = "IDLE"
+
+                    with m.State("HANDLE_GET_USER_BUTTON"):
+                        # take ownership of the interface
+                        m.d.comb += interface.claim.eq(1)
+
+                        # write the state of the user button into a local data register
+                        data = Signal(8)
+                        m.d.comb += data[0].eq(user_button)
+
+                        # transmit our data using a built-in handler function that
+                        # automatically advances the FSM back to the 'IDLE' state on
+                        # completion
+                        self.handle_simple_data_request(m, serializer, data)
+
+            return m
+
+
+
+When handling a control request in LUNA the first thing we look at is the ``setup.type`` field of the setup packet interface. We could check for other types such as ``USBRequestType.CLASS`` or ``USBRequestType.DEVICE`` if we wanted to implement handlers for them but, in this case, we're only interested in vendor requests.
+
+Next, we take ownership of the interface, in order to avoid conflicting with the standard, or other registered request handlers. Then we sequence the actual request handling with an `Amaranth Finite State Machine <https://amaranth-lang.org/docs/amaranth/latest/guide.html#lang-fsm>`__, starting in the  ``IDLE`` state.
+
+While in ``IDLE`` we wait for the ``setup.received`` signal to go high and signal the arrival of a new control request. We then parse the ``setup.request`` field to identify the next state to advance our FSM to. (We could also use the other setup packet fields such as ``wValue`` and ``wIndex`` for dispatch or as arguments but for now we're just intered in ``bRequest``.)
+
+The first handler of interest is ``UNHANDLED`` which simply stalls the interface in order to inform the host that it was an invalid request.
```

Comment:
`UNHANDLED` is not in the code anymore.

Reply:
🙈 intensifies. Tx @mndza, fixed in: https://github.com/greatscottgadgets/cynthion/pull/197/commits/214df318307849511fd7092a6097f44af0c9f0d9